### PR TITLE
CPLP-362: refactor provisioningDBAccess to use EFCore

### PR DESCRIPTION
**Pull request by ntruchsess**
- refactored ProvisioningDBAccess to use EFCore (affects creation of new Idps - used in useradministration and provisioning - plus Clients (only used in Provisioning))
- implemented CPLP-622: RegistrationService invite new User API endpoint using EFCore

### CODE REVIEW

**Comment by ntruchsess** [comment]
@qxz2mqe : this PR contains the EFCore-classes for your new table 'user_password_resets' in schema 'provisioning'. Please start with new branch from main as soon this PR is merged.

**Comment by ntruchsess** [comment]
@wira-mms : I did change table 'iam_users', renaming the pk back to iam_user_id to express that it (while technically being the pk of this table)  it refers to an external entity in keycloak. Changed the type to varchar(36) to match the keycloak column-type (we cannot know that is allways a guid as it may be set via the keycloak-api on user-creation)

**Comment by ntruchsess** [comment]
@evegufy : we need to allign on adjusting the configuration for catena-ng/Argo-CD deployment.

**Comment by ntruchsess** [comment]
@evegufy : secrets in keypass are updated in keepass for new 'dev' and also local devenv devhost/wsl. So in respect to new config this PR is ready to merge as soon secrets have been applied to new dev-environment

**Review by evegufy — APPROVED** [review]

**Review by wira-mms — APPROVED** [review]

File: src/registration/CatenaX.NetworkServices.Registration.Service/Controllers/RegistrationController.cs [hunk 1]
```diff
@@ -209,6 +208,23 @@ public async Task<IActionResult> SetCompanyWithAddressAsync([FromRoute] Guid app
                 return new StatusCodeResult((int)HttpStatusCode.InternalServerError);
             }
         }
+
+        [HttpPost]
+        [Authorize(Roles = "invite_user")]
+        [Route("application/{applicationId}/inviteNewUser")]
```
Comment (evegufy) [inline src/registration/CatenaX.NetworkServices.Registration.Service/Controllers/RegistrationController.cs:214]: @ntruchsess Just a note for after your/mine vacation: this new endpoint and the existing endpoint 'tenant/{tenant}/users' need be integrated.

**Pull request merged**